test/unit/test_id_storage.c

```diff
@@ -0,0 +1,62 @@
+#include <r_util.h>
+#include "minunit.h"
+
+
+static bool test_r_id_storage_prev_next_eq_0(void) {
+	// test if next reverts prev
+	ut32 id, _id;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	r_id_storage_add (ids, "aaa", &_id);
+	r_id_storage_add (ids, "bbb", &id);
+	r_id_storage_add (ids, "ccc", &_id);
+	_id = id;
+	r_id_storage_get_prev (ids, &_id);
+	r_id_storage_get_next (ids, &_id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, _id, "r_id_storage_{next/prev} reversal 0");
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_1(void) {
+	// test if next reverts prev (modulo wrap 1)
+	ut32 id, _id;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	r_id_storage_add (ids, "bbb", &id);
+	r_id_storage_add (ids, "aaa", &_id);
+	r_id_storage_add (ids, "ccc", &_id);
+	_id = id;
+	r_id_storage_get_prev (ids, &_id);
+	r_id_storage_get_next (ids, &_id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, _id, "r_id_storage_{next/prev} reversal 1");
+	mu_end;
+}
+
+static bool test_r_id_storage_prev_next_eq_2(void) {
+	// test if next reverts prev (modulo wrap 2)
+	ut32 id, _id;
+	RIDStorage *ids = r_id_storage_new (0, 15);
+	r_id_storage_add (ids, "aaa", &_id);
+	r_id_storage_add (ids, "ccc", &_id);
+	r_id_storage_add (ids, "bbb", &id);
+	_id = id;
+	r_id_storage_get_prev (ids, &_id);
+	r_id_storage_get_next (ids, &_id);
+	r_id_storage_free (ids);
+	
+	mu_assert_eq (id, _id, "r_id_storage_{next/prev} reversal 2");
+	mu_end;
+}
+
+static int all_tests(void) {
+	mu_run_test (test_r_id_storage_prev_next_eq_0);
```

Comment:
Could you add something like https://github.com/radareorg/radare2/pull/17271#issuecomment-683700546 please?

Reply:
Yes i can do that, the problem is that idpool doenst have such highest/lowest concept thing. this is implemented in io on top of the next/prev primitives. i can move that too